src/Apps/Partner/Components/PartnerViewingRooms/PartnerViewingRoomsGrid.tsx

```diff
@@ -0,0 +1,136 @@
+import * as React from "react"
+import { Box, Button, Column, GridColumns, Text } from "@artsy/palette"
+import {
+  RelayPaginationProp,
+  createPaginationContainer,
+  graphql,
+} from "react-relay"
+import { PartnerViewingRoomsGrid_viewingRoomsConnection$data } from "__generated__/PartnerViewingRoomsGrid_viewingRoomsConnection.graphql"
+import { extractNodes } from "Utils/extractNodes"
+import { ViewingRoomCardFragmentContainer } from "Apps/Partner/Components/PartnerViewingRooms/ViewingRoomCard"
+import { useState } from "react"
+
+interface PartnerViewingRoomsGridProps {
+  relay: RelayPaginationProp
+  viewingRoomsConnection: PartnerViewingRoomsGrid_viewingRoomsConnection$data
+  sectionTitle: string
+}
+
+export const PAGE_SIZE = 12
+
+const PartnerViewingRoomsGrid: React.FC<PartnerViewingRoomsGridProps> = ({
+  sectionTitle,
+  viewingRoomsConnection,
+  relay,
+}) => {
+  const viewingRoomNodes = extractNodes(
+    viewingRoomsConnection.viewingRoomsConnection
+  )
+
+  const [loading, setLoading] = useState(false)
```

Comment:
Relay pagination already has an internal loading state, you don't need your own, you can use `relay.isLoading()` in your pagination CTA.

Reply:
Tried it out - we still need the `useState` because `relay.isLoading()` returns `true` when the _previous_ call to `relay.loadMore()` [is still pending](https://relay.dev/docs/v10.1.3/pagination-container/#isloading), and returns `false` otherwise.

So if our loading state is just `loading={relay.isLoading()}`, we'll show the spinner forever after it's clicked the first time. And if it's `loading={!relay.isLoading()}`, it'll show a spinner forever on page load.

But this did make me notice that we should be checking in the click handler if we're still waiting on the previous call to resolve and stop trying to load more if so. Updated that!